### PR TITLE
Refactoring to specialised functional interfaces

### PR DESCRIPTION
**Pull request by ameyaKetkar**
Hello,
I am a Graduate Student at Oregon State University, studying the impact of autoboxing. I am testing a tool, which analyses the source code to find opportunities to refactor the code to a specialised implementation for primitives, thus eliminating autoboxing. All the changes, have been done manually, and have been verified by me. I have many more such patches, I will be happy to contribute.

### CODE REVIEW

**Comment by minborg** [comment]
Interesting. We have generated a numer of classes just for the reasons you mention. I will take a look at this!

**Comment by minborg** [comment]
Please have a look at https://github.com/speedment/speedment/blob/master/CONTRIBUTING.md to see how to contribute to the project.

**Comment by minborg** [comment]
How does the tool work. Is it a GitHub project?

**Comment by ameyaKetkar** [comment]
Sure. I ll create a new pull request against develop-3.0' branch. 
I have not github-ized it completely yet. It works as an eclipse plugin for now. 
You have to give mappings for classes. 
Eg. Function -> IntFunction/IntUnary/etc. according to conditions,
Map -> fastUtil maps, or gs-collection maps or whatever you choose.
So anything can be set in the mappings. Even the classes you have made to avoid autoboxing. 
In a nutshell,it analyses the autoboxing instances in the source-code, and refactors against a set of pre-conditions if mapping is available.

**Comment by ameyaKetkar** [comment]
We are working towards, inferring the mappings too.

**Comment by minborg** [comment]
Sounds cool! Nice to be able to detect this automatically. Presumably we have missed a number of them... Make sure to sign the contributor license agreement and you will get a "fame spot" in Speedment main POM file!

**Comment by ameyaKetkar** [comment]
Thanks a lot. 
If you could guide me to what classes you have made for avoiding autoboxing(probably, being used in the hot-path). I could probably test the too, and use it as an opportunity to continue contribution. I can see some in  `com.speedment.common.function` and  `com.speedment.common.function.collector`

**Comment by minborg** [comment]
Check out the module Speedment - Runtime - Field. There are many specialized implementations for different field types like IntField and LongField.

**Comment by minborg** [comment]
@ameyaKetkar Did you sign the CLA?

**Comment by ameyaKetkar** [comment]
Yes I did sign the CLA. 
Thank you, I will look into what you mentioned.

**Comment by minborg** [comment]
Added to pom: 
        <contributor>
            <name>Ameya Ketkar (GitHub:ameyaKetkar)</name>
            <timezone>America/Los_Angeles</timezone>
        </contributor>

**Comment by ameyaKetkar** [comment]
Thanks a lot.
